packages/inter-protocol/src/proposals/upgrade-vaults.js

```diff
@@ -150,9 +151,28 @@ export const upgradeVaults = async (powers, { options }) => {
   // Wait for at least one new price feed to be ready before upgrading Vaults
   void E.when(
     any(
-      Object.values(vaultBrands).map(brand =>
-        E(priceAuthority).quoteGiven(AmountMath.make(brand, 10n), istBrand),
-      ),
+      Object.values(vaultBrands).map(async brand => {
+        const getQuote = async lastRejectionReason => {
+          await null;
+          try {
+            return await E(priceAuthority).quoteGiven(
+              AmountMath.make(brand, 10n),
+              istBrand,
+            );
+          } catch (reason) {
+            if (
+              isUpgradeDisconnection(reason) &&
+              (!lastRejectionReason ||
+                reason.incarnationNumber >
+                  lastRejectionReason.incarnationNumber)
+            ) {
+              return getQuote(reason);
+            }
+            throw reason;
+          }
+        };
+        return getQuote(null);
+      }),
```

Comment:
What is the reason for waiting for a quote when upgrading `vaultFactory`? @mhofman

Reply:
> What is the reason for waiting for a quote when upgrading vaultFactory? @mhofman @Chris-Hibbert

When we run the coreEval to do the upgrade, we'll be upgrading vaultFactory, and replacing auctions, as well as replacing all the priceAuthorities. After replacing the priceAuthorities, the oracle operators have to manually intervene in order to accept new invitations, and do some reconfiguration in order to start feeding information to the new priceAuthorities. I don't have any indication of whether that will take a half hour or a couple of days.

During the interim, it's best that the vaults and auctions continue to listen to the old priceAuthorities. The indication that the switchover is complete is that any price starts coming via the new priceAuthorities. At that point, we upgrade the vaultFactory, which will then start paying attention to the new priceAuthorities.

Reply:
Thanks Chris!